quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py

```diff
@@ -0,0 +1,694 @@
+# MAGIC
+# MAGIC Once evaluation runs, click `View Evaluation Results` to open the MLflow UI for this Run.  This lets you:
+# MAGIC - See summary metrics
+# MAGIC - See root cause analysis that identifies the most important issues to fix
+# MAGIC - Inspect individual responses to gain intuition about how the Agent is performing
+# MAGIC - See the judge outputs to understand why the responses were graded as good/bad
+# MAGIC - Compare between multiple runs to see how quality changed between experiments
+# MAGIC
+# MAGIC You can also inspect the other tabs:
+# MAGIC - `Overview` lets you see the Agent's config/parameters
+# MAGIC - `Artifacts` lets you see the Agent's code
+# MAGIC
+# MAGIC This UIs, coupled with the speed of evaluation, help you efficiently test your hypotheses to improve quality, letting you reach the production quality bar in less time. 
+# MAGIC
+# MAGIC ![alt text](https://github.com/databricks/genai-cookbook/blob/demo_gifs/quick_start_demo/demo_gifs/eval_1.gif?raw=true "Title")
+
+# COMMAND ----------
+
+from mlflow.models.signature import ModelSignature
+from mlflow.models.rag_signatures import ChatCompletionResponse, ChatCompletionRequest
+from mlflow.models.resources import DatabricksServingEndpoint
+
+# First, we define a helper function so we can compare the agent across multiple parameters & LLMs.
+def log_and_evaluate_agent(agent_config: dict, run_name: str):
+
+    # Define the databricks resources so this logged Agent is deployment ready
+    resources = [DatabricksServingEndpoint(endpoint_name=agent_config["endpoint_name"])]
+
+    # Start a run to contain the Agent.  `run_name` is a human-readable label for this run.
+    with mlflow.start_run(run_name=run_name):
+        # Log the Agent's code/config to MLflow
+        model_info = mlflow.pyfunc.log_model(
+            python_model="fc_agent.py",
+            artifact_path="agent",
+            model_config=agent_config,
+            signature=ModelSignature(
+                inputs=ChatCompletionRequest(),
+                outputs=ChatCompletionResponse(),
+            ),
+            resources=resources,
+            input_example={"messages": [{"role": "user", "content": "What is lakehouse monitoring?"}]},
+            pip_requirements=["databricks-sdk[openai]", "mlflow", "databricks-agents"],
+        )
+
+        # Run evaluation
+        eval_results = mlflow.evaluate(
+            data=evals,  # Your evaluation set
+            model=model_info.model_uri,  # Logged Agent from above
+            model_type="databricks-agent",  # activate Mosaic AI Agent Evaluation
+        )
+
+        return (model_info, eval_results)
+
+
+# Now we call the helper function to run evaluation.
+# The configuration keys must match those defined in `fc_agent.py`
+model_info_gpt4o, eval_results = log_and_evaluate_agent(
```

Comment:
requires a deployed gpt-4o endpoint with name agents-demo-gpt4o in the workspace. do we want to use workspace foundation models like llama 3.1 405b or DBRX
will need to update variable name, endpoint name, and run name if the endpoint config changes